### PR TITLE
implement browser light / dark theme switching detection

### PR DESCRIPTION
**Pull request by pmario**
This PR fixes:  **[IDEA] Automatically Switch Between Dark / Light Palette Depending on Browser Setting** #7754 

- At the moment the setting can only be activated on browser startup. 
- Once this PR is merged, there can be new functionality that dynamically switches setting, when the browser themes are changed.

**The ControlPanel looks like this atm**

image

**If the Configuration Detection is on**

- Showing the configuration is animated, so users do not loose focus
- There should be enough explanatory text for users to understand what's going on
- The configuration detection got its own tiddler

image

**New Macros**

There are 2 new macros, that allow plugin authors and users to use the same UI as the core for configuring palette settings.

- `<<palette-picker tiddler:abc>>` ... Creates a dropdown dialogue that allows the user to define the tiddler which will contain the palette name. Defaults to: `$:/palette`
- `<<palette-info tiddler>>` ... Shows the selected palette swatches -> `$:/snippets/currpalettepreview`

**Internationalisation**

- All texts are translatable -- Wording can be improved. Just let me know.

### CODE REVIEW

**Comment by BurningTreeC** [comment]
Hi @pmario
Does this also take into account that the dark mode of the os can change while a wiki is open?

**Comment by pmario** [comment]
>Does this also take into account that the dark mode of the os can change while a wiki is open?

This setting has to be managed by the browser. If the browser is set to 

- "Automatic", it will use the setting from the OS. 
- Light -- The light theme is fixed even if OS is dark
- Dark -- The dark theme is fixed even if OS is light

So -- Yes

**FireFox**

image


**Edge**

image

File: core/wiki/macros/palette-picker.tid [hunk 1]
```diff
@@ -29,8 +29,15 @@ title: $:/core/macros/palette-picker
...
 	</$button>
-	<$reveal state=`$:/state/popup/$(tiddler)$` type="popup" position="belowleft" tag=div class="tc-drop-down" style="font-size:0.7em;">
-		<$transclude $tiddler="$:/snippets/paletteswitcher" palette=<<tiddler>> />
+	<$reveal state=`$:/state/popup/$(tiddler)$`
+		type="popup"
+		position="belowleft"
+		positionAllowNegative="yes"
+		tag="div"
+		class="tc-drop-down"
+		style=`$(style)$`
```
Comment (Jermolene) [inline core/wiki/macros/palette-picker.tid:38]: Is there a reason not to use `style=<<style>>`?

**Comment by pmario** [comment]
Sorry for wasting everyone's time. I'll close this UI centric PR and the related issue.

I'll create a new proposal related to the core function that detects the browser ["preferes-color-scheme" setting](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme).

**Comment by Jermolene** [comment]
@pmario I do not think that this work was a waste of time at all. It is a valuable experiment, and the feedback can help us iterate towards a better solution. I welcome your attention on this issue, it is definitely the kind of modernisation that I would like to be addressing at the moment

**Pull request closed without merging**